### PR TITLE
search ui: make sidebar collapsible

### LINKED ISSUE #37552: Make sidebar on search results page collapsible
Currently, the sidebar on the search results page persists and is not collapsible. We have a new design that moves the sidebar to the right, makes it collapsible, and uses more natural language to describe the filters.

image

This is inconsistent with other sidebar UX patterns in the app, and many users find this sidebar to be unhelpful or annoying.

We should:
- Make the sidebar hideable via the same mechanism employed in the file tree
- Persist the users preference in temporary settings to it's not reset when they leave the page

## Designs
[Prototype](https://www.figma.com/proto/8MkIcpDtjv6tqgxZFcEQDt/Core-workflow-punch-list?node-id=176%3A14490&scaling=scale-down&page-id=0%3A1&starting-point-node-id=176%3A19861&show-proto-sidebar=1)

[Figma](https://www.figma.com/file/8MkIcpDtjv6tqgxZFcEQDt/Core-workflow-punch-list?node-id=176%3A14490)

### PR DESCRIPTION
**Pull request by limitedmage**
Closes #37552

Requires "Simple UI" toggle on.

- Makes sidebar collapsible using a sidebar button strip to toggle the filters
- Updates the sidebar to match the new style.

## Test plan

Manual test matrix:
- Dark/light mode
- Mobile/full UI
- Simple UI on/off

https://user-images.githubusercontent.com/206864/179631038-6c55ce70-d161-45ce-b9b5-7931c97029f2.mp4

No changes should appear when the "simple UI" toggle is off and visual tests should not have any changes.

## App preview:

- [Web](https://sg-web-jp-collapsablesidebar.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-lhszwdxljz.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/index.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 export * from './input/toggles'
 export * from './input/useQueryIntelligence'
-export * from './results/StreamingSearchResultsList'
 export * from './results/progress/StreamingProgress'
 export * from './results/progress/StreamingProgressCount'
 export * from './results/sidebar/FilterLink'
-export * from './results/sidebar/SearchSidebar'
 export * from './results/sidebar/revisions'
+export * from './results/sidebar/SearchSidebar'
+export * from './results/sidebar/SidebarButtonStrip'
```
Comment (limitedmage) [inline client/search-ui/src/index.ts:15]: This is the only new one, the rest have been sorted alphabetically.

File: client/search-ui/src/results/sidebar/SearchSidebar.tsx [hunk 2]
```diff
@@ -234,14 +236,21 @@ export const SearchSidebar: React.FunctionComponent<React.PropsWithChildren<Sear
...
         <aside className={classNames(styles.sidebar, props.className)} role="region" aria-label="Search sidebar">
-            {props.prefixContent}
-            {body}
+            <StickyBox className={styles.stickyBox} offsetTop={8}>
+                <div className={styles.header}>
+                    <Button variant="icon" onClick={() => setSelectedTab(null)}>
+                        <Icon svgPath={mdiClose} aria-label="Close sidebar" />
+                    </Button>
+                </div>
+                {props.prefixContent}
```
Comment (limitedmage) [inline client/search-ui/src/results/sidebar/SearchSidebar.tsx:251]: This also fixes the issue with odd scrolling when the onboarding tour is enabled, by moving the container of the tour into the StickyBox

**Review by rvantonder — APPROVED** [review]
I approve because I looked at the video. any bugs / design can be dealt with in follow up PRs :D

**Comment by limitedmage** [comment]
@benvenker The dual-pane is the existing design we have for mobile views, this doesn't change it. I just wanted to make sure this change didn't break it :)

**Comment by limitedmage** [comment]
Due to popular demand, I'll merge now, but will happily address any further feedback in subsequent PRs :)

**Comment by almeidapaulooliveira** [comment]
Beatiful!

**Pull request merged**